tests/study_tests/test_study.py

```diff
@@ -1166,6 +1166,21 @@ def objective(trial: Trial) -> list[float]:
         assert len(trial.values) == n_objectives
 
 
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_best_trial(storage_mode: str) -> None:
+    with StorageSupplier(storage_mode) as storage:
+        study = create_study(storage=storage)
+        with pytest.raises(ValueError):
+            study.best_trial
+        study.tell(study.ask(), 1)
+        assert study.best_trial.number == 0
+        study.tell(study.ask(), 0)
+        assert study.best_trial.number == 1
+        study.tell(study.ask(), 0)
+        # If the objective values are the same, the best trial will be the first trial executed.
+        assert study.best_trial.number == 1
```

Comment:
[Question] I believe this is merely an implementation detail. Do you think there are any advantages to making this behavior a specification?